### PR TITLE
[Gekidou MM-44517] Add commonmark to search results

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR adds highlighting for search terms in posts. 

Test Searches (**on Community**)
- [x] `Test "but most"` - highlights `Test` and `but most` phrase
- [x] `"but most" Test` - highlights `Test` and `but most` phrase 
- [x] `"but most"` - highlights `but most`

This results from the search result which includes posts that don't have matching terms (webapp gets the same result)
image

Differences between Webapp and Mobile:
| webapp | mobile |
| -- | -- | 
| image | image |

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44517

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

File: app/components/markdown/transform.ts [hunk 1]
```diff
@@ -160,6 +160,105 @@ export function highlightMentions(ast: Node, mentionKeys: UserMentionKey[]) {
     return ast;
 }
 
+const puncStart = /^[^\p{L}\d\s#]+/u;
+const puncEnd = /[^\p{L}\d\s]+$/u;
+
+export function parseSearchTerms(searchTerm: string): string[] {
```
Comment (jfrerich) [inline app/components/markdown/transform.ts:166]: take from https://github.com/mattermost/mattermost-webapp/blob/master/utils/text_formatting.tsx#L677

File: app/components/markdown/transform.ts [hunk 2]
```diff
@@ -160,6 +160,105 @@ export function highlightMentions(ast: Node, mentionKeys: UserMentionKey[]) {
...
+        if (term.charAt(term.length - 1) !== '*') {
+            term.replace(puncEnd, '');
+        }
+        return term;
+    });
+
+    return terms;
+}
+
+function convertSearchTermToRegex(term: string): SearchPattern {
```
Comment (jfrerich) [inline app/components/markdown/transform.ts:231]: Taken from https://github.com/mattermost/mattermost-webapp/blob/master/utils/text_formatting.tsx#L742

**Comment by enahum** [comment]
@jfrerich any progress on this one? just making sure it does not fall under the radar.

**Comment by jfrerich** [comment]
@enahum, I just merged latest gekidou into this branch and have it working again. 

@hmhealey, can you take a look at this PR provide your thoughts? All I did was copy some code from webapp (`convertSearchTermToRegex` and `parseSearchTerms`) and pass the results into our `Post` component.

**Review by hmhealey — COMMENTED** [review]
I swore we added this in https://github.com/mattermost/mattermost-mobile/pull/6303, but I guess we just added half that feature since we didn't have proper search yet on mobile.

This looks pretty good to me minus the couple questions I had. Hopefully we'll get that Markdown code packaged together some day so that we don't have to duplicate it between mobile and web

**Comment by enahum** [comment]
@jfrerich when you can please take a look at @hmhealey comments

**Comment by jfrerich** [comment]
@enahum, I am out this week and will get back into this PR soon with answers for Harrison!

**Comment by larkox** [comment]
@jfrerich When you get back to this ticket, it would be great if you could take a look at this conversation (The part regarding the todo on the markdown) https://community.mattermost.com/core/pl/3e8bsng5m3bf5yxpm8eegzd38c

The main idea is to check if the highlight of the search works well in certain scenarios, and if so, remove the todo on `app/components/markdown/transform.ts`

**Comment by enahum** [comment]
closing for now

**Pull request closed without merging**